artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/AMQPBrokerConnection.java

```diff
@@ -239,7 +271,44 @@ SimpleString getMirrorSNF(AMQPMirrorBrokerConnectionElement mirrorElement) {
       return snf;
    }
 
+   /**
+    * Adds a remote link closed event intercepter that can intercept the closed event and if it
+    * returns true indicate that the close has been handled and that normal broker connection
+    * remote link closed handling should be ignored.
+    *
+    * @param id
+    *    A unique Id value that identifies the intercepter for later removal.
+    * @param intercepter
+    *    The predicate that will be called for any link close.
+    *
+    * @return this broker connection instance.
+    */
+   public AMQPBrokerConnection addLinkClosedIntercepter(String id, Predicate<Link> intercepter) {
```

Comment:
Intercepter -> Interceptor for consistency with other areas?

Reply:
fixed